src/modality-specific-files/task-events.md

```diff
@@ -175,10 +175,47 @@ This can be done by using a `/stimuli` directory or by reference to a stimuli da
 The stimulus files can be added in a `/stimuli` directory
 (under the root directory of the dataset; with optional subdirectories) AND using a
 `stim_file` column in `events.tsv` mentioning which stimulus file was used
-for a given event,
+for a given event.
 
-There are no restrictions on the file formats of the stimuli files,
-but they should be stored in the `/stimuli` directory.
+It is RECOMMENDED to include a `stimuli.tsv` table containing a list of the stimuli present in the `/stimuli` directory.
+The `stimuli.tsv` would list the stimulus files as rows and provide more information about the stimulus files as columns, such as their types (for example, still images, movies, audio, multi-channel audio, etc), text descriptions, and optional columns including references, relevant HED tags, etc.
+Annotations for the still stimulus files can be properly held in `stimuli.tsv` as specific columns. The `stimuli.tsv` file can be accompanied by a `stimuli.json` file describing the columns in detail. The `stimuli.json`, similar to the `events.json` file, MAY also include additional information about the stimuli, such as references to databases, HED tags, etc.
+The `stimuli.tsv` MUST include the following columns:
+
+{{ MACROS___make_columns_table(
+   {
+      "stim_file": ("REQUIRED", "There are no restrictions on the file formats of the stimuli files, but they should be stored in the `/stimuli` directory (under the root directory of the dataset; with OPTIONAL subdirectories). The values under the stim_file column correspond to a path relative to `/stimuli`. For example `images/cat03.jpg` will be translated to `/stimuli/images/cat03.jpg`. This column MUST appear as the first column in the file."),
+      "type": ("REQUIRED", "String. One of `movie`, `still_image`, `audio`, `annotation`, or Freeform.")
+   }
+) }}
+
+### Time-varying stimuli
+A single-line annotation in a `stimuli.tsv` column would be sufficient for still stimulus files (for example, images). However, a time-varying stimulus file could need separate *temporal annotations* for every frame (that is, the smallest temporal resolution of the stimulus file).
+
+The temporal annotations for time-varying stimuli can be stored in a separate file, `annot-XXX_stimulus.tsv`, which would hold the annotations for time points of the time-varying stimulus similar to an `events.tsv` file. `stimulus.tsv` MUST follow the same format and contain the same REQUIRED and OPTIONAL columns of the `events.tsv` file, and MAY be accompanied by a JSON file describing the columns in detail.
```

Comment:
I might be misreading this but it seems here that the `annot-*_stimulus.tsv` file is not required to be associated with a specific file in the `stimuli` directory (for example "the-present_movie.mp4"). This might make synchronization harder in the future?